### PR TITLE
Added SequencePhenotype class 

### LINKED ISSUE #2: Enable SequencePhenotype to work with PhenotypeFactory
### Description
Currently, `PhenotypeFactory` does not support `SequencePhenotype`. We want to enable this.
Optional: do we want to be able to create arbitrary phenotypes?

### Possible Solution

- Need static string data member to represent the new sequence phenotype
- Add an appropriate if statement to` PhenotypeFactory.makeVirusPhenotype()`
- Add an appropriate if statement to `PhenotypeFactory.makeHostPhenotype()`

### PR DESCRIPTION
**Pull request by thienktran**
## Description

Created a data structure, `SequencePhenotype`, that implements the Phenotype interface. `SequencePhenotype` allows us to represent viruses as a sequence of `Strings` rather than `doubles`.

Major differences between `SequencePhenotype` and other implementations of the Phenotype interface
- Distances between two `SequencePhenotype`s are calculated using hamming distance.
- Point mutation is chosen randomly - all indices have an equal chance of mutating to A, G, T, or C. 

Closes #2

## Tests

Tested

- Empty constructor and the constructor for a simple sequence 
- Hamming distance between sequences 
- Getter and setter methods
- Substitution (random point mutation) 
- toString
- equals

Wrote a representation invariant and made a method to confirm that it's never gets broken when DEBUG == true during program runs. 

## Checklist:

* [x] The code uses informative and accurate variable and function names
* [x] The functionality is factored out into functions and methods with logical interfaces
* [x] Comments are up to date, document intent, and there are no commented-out code blocks
* [x] Commenting and/or documentation is sufficient for others to be able to understand intent and implementation
* [x] TODOs have been eliminated from the code
* [x] The corresponding issue number (e.g. `#278`) has been searched for in the code to find relevant notes
* [ ] Documentation has been redeployed

### CODE REVIEW

File: Host.java [hunk 1]
```diff
@@ -61,6 +61,8 @@ public void addToHistory(Phenotype p) {
 		}
 		newHistory[immuneHistory.length] = p;
 		immuneHistory = newHistory;
+		//System.out.print("P: " + p + "      ");
```
Comment (matsen) [inline Host.java:64]: Cleanup

File: Parameters.java [hunk 2]
```diff
@@ -75,7 +75,8 @@
 	public static double sdStep = 0.3; 
 	public static boolean mut2D = false;						// whether to mutate in a full 360 degree arc
 	public static boolean fixedStep = false;					// whether to fix mutation step size
-		
+	public static String sequence = "AGTC";
```
Comment (matsen) [inline Parameters.java:78]: startingSequence?

File: parameters.yml [hunk 3]
```diff
@@ -51,7 +51,7 @@ demeAmplitudes: [0.1, 0., 0.1]              # amplitude of seasonality
 demeOffsets: [0., 0., 0.5]                  # seasonal offset relative to the year
     
                                             # immune parameters  
-phenotypeSpace: "geometric"                 # options include: geometric, geometric3d, geometric10d
+phenotypeSpace: "sequence"                 # options include: geometric, geometric3d, geometric10d, sequence
```
Comment (matsen) [inline parameters.yml:54]: Do we want to keep the original simulation parameter set but provide a separate new one?

File: Simulation.java [hunk 4]
```diff
@@ -13,19 +13,19 @@
 	private double tmrca;
 	private double netau;
 	private double serialInterval;
-	private double antigenicDiversity;	
-	
+	private double antigenicDiversity;
```
Comment (matsen) [inline Simulation.java:16]: Can you reformat the entire repo once so we get all of these non-functional changes out of the way?

**Review by matsen — COMMENTED** [review]
Just a quick once over. Let's look at it in more detail in person.

File: parameters.yml (hunk 3 above)
Comment (thienktran) [reply parameters.yml:54]: That's a good idea - thanks!

File: TestSequencePhenotype.java [hunk 5]
```diff
@@ -111,4 +111,18 @@ public void testToString() {
...
+	 */
+	@Test
+	public void testEquals() {
+		// Same SequencePhenotype objects are equal
+		assertTrue(emptyPheno.equals(emptyPheno));
+
+		SequencePhenotype simplePhenoTest = new SequencePhenotype("ACGT");
+
+		// SequencePhenotype objects with the same sequence are equal
+		assertTrue(simplePheno.equals(simplePhenoTest));
```
Comment (zorian15) [inline TestSequencePhenotype.java:125]: Maybe a test to ensure `SequencePhenotype.equals()` returns false whenever the sequences are different?

File: SequencePhenotype.java [hunk 6]
```diff
@@ -1,50 +1,107 @@
...
+ */
 public class SequencePhenotype implements Phenotype {
 
-    // constants
-    @SuppressWarnings("unchecked")
-    public final String[] NUCLEOBASES = new String[]{"A", "C", "G", "T"};
+    /**
+     * The valid letters that make up the sequence of this SequencePhenotype
+     */
+    public final String[] NUCLEOTIDES = new String[]{"A", "C", "G", "T"};
```
Comment (zorian15) [inline SequencePhenotype.java:15]: Maybe we should think about renaming this something like `ALPHABET` or `AMINO_ACIDS` if we're going to be leaving the nucleotide representation in the future (idk if we'll keep it around and have an option for either DNA or AAs). But doesn't need to be done until we get ready to cross that bridge.

File: SequencePhenotype.java [hunk 7]
```diff
@@ -1,50 +1,107 @@
-import java.util.*;
+import java.lang.*;
+
+/**
+ * <b>SequencePhenotype</b> represents a phenotype.
+ * SequencePhenotypes are identified by their data contents.
+ * SequencePhenotype permits all data, except for <b>""</b> and <b>null</b>.
+ *
+ * @author Thien Tran
+ */
```
Comment (zorian15) [inline SequencePhenotype.java:9]: I love all of the docstrings you added to this class! 🎸

**Review by zorian15 — COMMENTED** [review]
I think this is nice code @thienktran -- a few minor things

**Review by matsen — APPROVED** [review]

File: SequencePhenotype.java [hunk 8]
```diff
@@ -1,50 +1,95 @@
...
 public class SequencePhenotype implements Phenotype {
 
-    // constants
-    @SuppressWarnings("unchecked")
-    public final String[] NUCLEOBASES = new String[]{"A", "C", "G", "T"};
+    /**
+     * The valid letters that make up the sequence of this SequencePhenotype
+     */
+    public final char[] ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".toCharArray();
```
Comment (zorian15) [inline SequencePhenotype.java:15]: I thought we were moving to an amino-acid representation -- are we allowing all English alphabet now?

**Review by zorian15 — APPROVED** [review]
Just one quick Q but I approve as well

**Review by zorian15 — APPROVED** [review]

**Pull request merged**